### PR TITLE
[AIRFLOW-3458] Move models.Connection into separate file

### PR DESCRIPTION
**Pull request by BasPH**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-3458
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.

AIRFLOW-3458

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

This PR:
1. Creates a module airflow.models
2. Moves /airflow/models.py to /airflow/models/\_\_init__.py. This results in no breaking changes moving /airflow/models.py.
3. Moves models.Connection from models.py into a separate file /airflow/models/connection.py and is the first of many classes to split away from models.py.

Goal is to split models.py into manageable and logical files. Also see [discussion on mailing list](https://lists.apache.org/api/atom.lua?mid=3c15a871b15f99b4dd4bef25085b4de1c167c728f08262df73ef7217@%3Cdev.airflow.apache.org%3E). This is the first PR of many in the list of classes to extract from models.py.

Note I moved /airflow/models.py to /airflow/models/\_\_init__.py, this is a *temporary* solution as suggested on the mailing list to extract all classes one by one while not inducing any breaking changes.

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

No new functionality.

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

No new functionality.

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

File: tests/contrib/hooks/test_azure_cosmos_hook.py [hunk 1]
```diff
@@ -1,202 +1,202 @@
...
-        self.cosmos = AzureCosmosDBHook(azure_cosmos_conn_id='azure_cosmos_test_key_id')
-        self.cosmos.delete_collection(self.test_collection_name)
-        expected_calls = [mock.call().DeleteContainer('dbs/test_database_default/colls/test_collection_name')]
-        cosmos_mock.assert_any_call(self.test_end_point, {'masterKey': self.test_master_key})
-        cosmos_mock.assert_has_calls(expected_calls)
-
-
-if __name__ == '__main__':
-    unittest.main()
+# -*- coding: utf-8 -*-
```
Comment (Fokko) [inline tests/contrib/hooks/test_azure_cosmos_hook.py:1]: Looks like windows line-endings

**Review by Fokko — APPROVED** [review]
Awesome work @BasPH 

Thanks for picking this up!

**Comment by ashb** [comment]
@BasPH @Fokko I'd like to cherry-pick this in to 1.10.3, but do do that we need a deprecation shim.

I can do this with a few lines at the _end_ of models/__init__.py:

```python
# To avoid circular import on Python2.7 we need to define this at the _bottom_
from zope.deprecation import deprecated
from airflow.models.connection import Connection  # noqa

deprecated('Connection', 'has been moved to airflow.models.connection package. Please update your code before Airflow 2.0')

```

Are you happy for me to "just" do that on the release branch?

Trying that https://github.com/ashb/airflow/commit/f6d200876c9ab41aa68e8d51bb1e56aa38720393  - tests running https://travis-ci.org/ashb/airflow/builds/509059430

**Comment by BasPH** [comment]
@ashb sounds good to me!

**Pull request merged**